### PR TITLE
Update publish shell workflow to allow for independent release of either shell or creators packages

### LINKED ISSUE #11694: Update publish shell workflow to allow for independent release of either shell or creators packages
Update publish shell workflow to allow for independent release of either shell or creators packages.

Right now, the workflow works on a tagged release and it releases both packages to NPM with the same version number.

We should be able to target them individually and each with their own semver version numbering.

### PR DESCRIPTION
**Pull request by aalves08**
### Summary
Fixes #11694 

### Occurred changes and/or fixed issues

- add step for checking tag version matching on publish shell workflow -> `check-package-tag-version.sh`
- update `publish-shell` script to allow for independent publish of either `Shell` or `Creators` packages to NPM
- remove unused `TMP_DIR` var in `publish-shell` script
- update creators package version to match what was published on NPM

### Areas or cases that should be tested

- you can run a check for the `check-package-tag-version.sh` script by running it manually with a `TAG` of your choice, such as:
```
export TAG=shell-pkg-v2.0.1 && .github/workflows/scripts/check-package-tag-version.sh
```

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

**Comment by aalves08** [comment]
FYI @richard-cox @nwmac

File: shell/scripts/publish-shell.sh [hunk 1]
```diff
@@ -93,10 +103,40 @@ function publish() {
...
+    publish "Extension creator" ${PKG_DIST}/extension/ ${PKG_V}
+  fi
+else
+  # other workflows...
+  # Publish the packages - don't tag the git repo and don't auto-increment the version number
+  publish "Shell" ${SHELL_DIR}
+  publish "Extension creator" ${PKG_DIST}/extension/
+
+  echo "Done"
+fi
```
[the remark this answers is not in the document]
Comment (aalves08) [reply shell/scripts/publish-shell.sh]: I think you've got a point here... 🤔 

I don't think it's called anywhere else and we can simplify it and adjust the `test-plugins-build` script. I've done that, let's see if it passes the gates

File: .github/workflows/scripts/check-package-tag-version.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    SHELL_VERSION=$(jq -r .version ${SHELL_DIR}/package.json)
+    if [ "$SHELL_VERSION" == "$PKG_VERSION" ]; then
+      echo "tag check: shell versions match"
+      exit 0
+    else 
+      echo "Version mismatch for the shell package publish => shell: ${SHELL_VERSION} vs tag: ${PKG_VERSION}. Please redo the tagging properly"
+      exit 1
+    fi
+    ;;
+  creators)
```
Comment (jordojordo) [inline .github/workflows/scripts/check-package-tag-version.sh:33]: `shell` and `creators` should be quoted.

```bash
"shell")
...
"creators")
...
```

File: shell/scripts/publish-shell.sh [hunk 3]
```diff
@@ -93,10 +81,27 @@ function publish() {
...
+echo "PKG_NAME ${PKG_NAME}"
+echo "PKG_V ${PKG_V}"
+
+# version comparison checks
+case $PKG_NAME in
+  shell)
+    echo "Publishing only Shell pkg via tagged release"
+    publish "Shell" ${SHELL_DIR} ${PKG_V}
+    ;;
+  creators)
```
Comment (jordojordo) [inline shell/scripts/publish-shell.sh:99]: Same here with the quotes for shell and creators

**Review by jordojordo — CHANGES_REQUESTED** [review]

**Review by jordojordo — APPROVED** [review]

**Review by nwmac — APPROVED** [review]

**Comment by nwmac** [comment]
This needs to get in - some e2e tests are not passing, but after review, the failures there do not appear to be related to the changes here - getting this in as its needed to unblock other PRs.

**Pull request merged**